src/dotty/tools/dotc/transform/Erasure.scala

```diff
@@ -263,6 +263,12 @@ object Erasure extends TypeTestsCasts{
           else
             cast(tree, pt)
       }
+
+
+    /** Is `tpe` a type which is a bottom type for Dotty, but not a bottom type for JVM */
+    def isNonJVMBottomType(tpe: Type)(implicit ctx: Context): Boolean = {
+      tpe.derivesFrom(ctx.definitions.NothingClass) || tpe.derivesFrom(ctx.definitions.NullClass)
+    }
   }
```

Comment:
Definitions#isBottomType(tp) is defined to be `tp.symbol \in {NothingClass, NullClass}` 
It does not work for types that do not return a usable symbol, such as And\Or types.